packages/dev/inspector-v2/src/components/properties/frameGraph/frameGraphGeneralProperties.tsx

```diff
@@ -0,0 +1,28 @@
+import type { FrameGraph } from "core/index";
+
+import type { FunctionComponent } from "react";
+
+import { CheckboxPropertyLine } from "shared-ui-components/fluent/hoc/propertyLines/checkboxPropertyLine";
+import { ButtonLine } from "shared-ui-components/fluent/hoc/buttonLine";
+
+export const FrameGraphGeneralProperties: FunctionComponent<{ frameGraph: FrameGraph }> = (props) => {
+    const { frameGraph } = props;
+
+    return (
+        <>
+            <CheckboxPropertyLine
+                label="Optimize texture allocation"
+                description="Whether to optimize texture allocation."
+                value={frameGraph.optimizeTextureAllocation}
+                onChange={(checked) => (frameGraph.optimizeTextureAllocation = checked)}
+            />
+            {frameGraph.scene.frameGraph !== frameGraph && <ButtonLine onClick={() => (frameGraph.scene.frameGraph = frameGraph)} label="Set as scene's frame graph" />}

```

Comment:
You can add something like `const isSceneFrameGraph = useProperty(frameGraph.scene, "frameGraph");

and then

`{!isSceneFrameGraph && <ButtonLine ... />}`

This way if any other code re-assigns the scene's frameGraph property (such as a PG), this UI will correctly update in real time.